pkg/dns/azure/client/client.go

```diff
@@ -94,19 +94,16 @@ func newRecordSetClient(config Config, userAgentExtension string) (*recordSetCli
 	return &recordSetClient{client: rc}, nil
 }
 
-func (c *recordSetClient) Put(ctx context.Context, zone Zone, arec ARecord) error {
+func (c *recordSetClient) Put(ctx context.Context, zone Zone, arec ARecord, metadata map[string]*string) error {
 	rs := dns.RecordSet{
 		RecordSetProperties: &dns.RecordSetProperties{
 			TTL: &arec.TTL,
 			ARecords: &[]dns.ARecord{
 				{Ipv4Address: &arec.Address},
 			},
+			Metadata: metadata,
 		},
 	}
-	if arec.Label != "" {
```

Comment:
This is no longer required, as the metadata passed to this method, has the cluster_id tag which is being added here.

Reply:
It's notable that `func (c *privateRecordSetClient) Put` didn't have this code, so the operator wasn't setting this label for private records, but now it will do so.  It seems you're gifting us a free bug-fix as part of this new feature.  Thanks!

Reply:
But is a privateRecordSet supposed to be marked as "owned" ?

Reply:
As part of [Azure tag support](https://issues.redhat.com/browse/CORS-2249) feature, ocp and user defined tags are added to all possible resources created by OCP.